### PR TITLE
ISSUE-121 PGP - EncryptedEmailDetailedView/get Method

### PR DESCRIPTION
**Pull request by vttranlina**
ref https://github.com/linagora/tmail-backend/issues/121

### CODE REVIEW

**Comment by vttranlina** [comment]
`LinagoraEncryptedEmailDetailedViewGetMethodContract#methodShouldSupportBackReference` is not pass.
Reason: https://github.com/linagora/tmail-backend/pull/187

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEncryptedEmailDetailedViewGetMethodContract.scala [hunk 1]
```diff
@@ -0,0 +1,1003 @@
...
+
+    server.getProbe(classOf[MailboxProbeImpl])
+      .createMailbox(BOB_INBOX_PATH)
+
+    uploadPublicKey()
+  }
+
+  def randomMessageId: MessageId
+
+  private def uploadPublicKey(): Unit = {
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEncryptedEmailDetailedViewGetMethodContract.scala:78]: We could share that accross tests by moving it to a Fixture (Don't repeat yourself)

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEncryptedEmailDetailedViewGetMethodContract.scala [hunk 2]
```diff
@@ -0,0 +1,1003 @@
...
+      .inPath("methodResponses[0][1]")
+      .isEqualTo(
+        s"""{
+           |    "accountId": "$ACCOUNT_ID",
+           |    "state": "${INSTANCE.value}",
+           |    "list": [
+           |        [
+           |            "${messageId.serialize()}",
+           |            {
+           |                "encryptedPreview": "$${json-unit.ignore}",
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEncryptedEmailDetailedViewGetMethodContract.scala:433]: Convenient trick :+1:  <3

**Review by chibenwa — COMMENTED** [review]
Awesome work !!!

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEncryptedEmailDetailedViewGetMethodContract.scala (hunk 2 above)
Comment (Arsnael) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEncryptedEmailDetailedViewGetMethodContract.scala:433]: That just ignore the value here? Yeah that's awesome, good finding :)

**Review by Arsnael — COMMENTED** [review]
Read it, nothing to add, good job overall :)

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**